src/main/java/com/pretz/dyntreeserver/controller/AuthController.java

```diff
@@ -21,25 +23,13 @@ public AuthController(UserService userService) {
 
     @RequestMapping(method = POST, path = "/auth")
     public ResponseEntity<String> login(@RequestBody User user) {
-        boolean userExists = userService.validateUser(user);
-        if (userExists) {
-            //is it RESTful?
-            UserSession userSession = new UserSession(user);
-            return new ResponseEntity<>(HttpStatus.OK);
-        }
-        else {
-            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
-        }
+        String userToken = userService.validateUser(user);
+        return new ResponseEntity<>("User " + user.getName() + " successfully logged in! \nJWT Token: " + userToken, new HttpHeaders(), HttpStatus.OK);
     }
 
     @RequestMapping(method = POST, path = "/create_user")
     public ResponseEntity<String> createUser(@RequestBody User user) {
```

Comment:
is this `User` class here the Model class, same as the one used in `Repository`?

Reply:
If yes, I'd suggest introducing [DTO classes](https://stackoverflow.com/questions/43820194/should-i-consider-using-dto-for-spring-rest-controller-layers-instead-of-entitie) ([more](https://auth0.com/blog/automatically-mapping-dto-to-entity-on-spring-boot-apis/))

Reply:
There is UserRepo class which extends Spring JpaRepository class. As far as I understand, it is the same as DAO object. It handles database operations regarding User. User is Model (and @Entity class). Why I need another type of objects?

Reply:
Because you don't want to handle Entities over to the View layer (as justified in the first link).

e,g, you don't want Spring to invoke hibernate related logic (that is - database stuff) every time something hits an endpoint. Usually it's a good idea to split data, so that `DTOs` are passed between Controllers and Services whereas `Entities` are passed between Services and Repositories.

Reply:
Of course, it's [not](https://stackoverflow.com/questions/36174516/rest-api-dtos-or-not) always [necessary](http://guntherpopp.blogspot.com/2010/09/to-dto-or-not-to-dto.html)

Reply:
Ok, I understand now